File: personal/homework/mynetty-python/PersonServiceImpl.py
from py.thrift import ttypes
from py.thrift import PersonService


class PsersonServiceImpl(PersonService.Iface):

    def getPersonByUsername(self, username):
        p = ttypes.Person()
        p.married = True
        p.username = "王连成"
        p.age = 30
        return p

    def savePerson(self, p):
        print(p)

File: personal/homework/mynetty/src/main/java/com/learn/netty/thrift/PersonServiceImpl.java
package com.learn.netty.thrift;

import org.apache.thrift.TException;

public class PersonServiceImpl implements PersonService.Iface{
    @Override
    public Person getPersonByUsername(String username) throws DataException, TException {
        Person person = new Person();
        person.setAge(12)
                .setMarried(true)
                .setUsername(username);
        return person;
    }

    @Override
    public void savePerson(Person person) throws DataException, TException {
        System.out.println(person.toString());
    }
}
